### ISSUE TITLE
[Bug] When tables get too big, the serializer panics with an uninformative error message

### ISSUE THREAD

**Issue by brmataptos** [aptos-labs/aptos-core#9802]
# 🐛 Bug

Greg reports :

> I’ve successfully hit this confusing error message, and the number 456 seems to change when I make some functions not inline (I’ve got inlines, in inlines, inlines for readability purposes)
> thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: value (456) cannot exceed (255)', third_party/move/move-compiler/src/compiled_unit.rs:200:18

While this likely is related to excessive inlining causing the local function table to be too big, this error message is generated inside the serializer code in write_as_uleb128 in move-binary-format/src/serializer.rs.  We should fix the error message to include some indication of what is overflowing, and then we have a better idea of what needs to be fixed.

Ideally, this would be turned into a useful end-user message.

## To reproduce

Need test cases to reproduce.

## Expected Behavior

At least a more informative Panic message, e.g. by adding: "in serialize_field_inst_index".

Better: a useful end-user message that suggests how it might be solved ("break up your function into pieces or reduce inlining").

**Comment by wrwg** [comment]
This is already dealt with in compiler v2. It has to be a real compiler error message, the serializer is not in the position to emit such messages.

**Comment by brmataptos** [comment]
> This is already dealt with in compiler v2. It has to be a real compiler error message, the serializer is not in the position to emit such messages.

I don't think this is true.  (See https://github.com/aptos-labs/aptos-core/pull/12593).  Should we make it so?  @wrwg
